### PR TITLE
[MERGE] mail, mass_mailing, test_*: improve reply-to management

### PR DESCRIPTION
**Pull request by sht-odoo**
PURPOSE

We receive a lot of tickets from people who set a reply-to on their email
templates and are surprised that it doesn't work.  Indeed this field is used
for mass mailing emails and templates but not in regular Chatter.

SPECIFICATIONS

Update reply-to field placeholder to "Preferred email address when sending
via mass mailing options".

Update reply-to field helper message to "Preferred email address when sending
via mass mailing options. <br> Only used when the answer is not added into
the original discussion.""

Update the no_auto_thread field label to "Reply to" in composer and introduce
a new radio button replacing the checkbox

  * The original discussion (update)
  * Another email address (new)

Rename fields on mail_thread and wizard: ``no_auto_thread`` should be replaced
to ``reply_to_force_new`` to ease understanding and be prefixed by reply_to.

Propagate radio keys (update and new) to mass mailing in order to have a
coherent naming (was thread and email). This naming is also coherent with
gateway naming (message_update and message_new).

LINKS

Task ID-2117639
COM PR odoo/odoo#40931
ENT PR odoo/enterprise#17941
UPG PR odoo/upgrade#2419

### CODE REVIEW

File: addons/mail/wizard/mail_compose_message.py [hunk 1]
```diff
@@ -126,9 +126,11 @@ def default_get(self, fields):
...
     # destination
-    reply_to = fields.Char('Reply-To', help='Reply email address. Setting the reply_to bypasses the automatic thread creation.')
-    no_auto_thread = fields.Boolean(
-        'No threading for answers',
+    reply_to = fields.Char('Reply-To-Email', help='Reply email address. Setting the reply_to bypasses the automatic thread creation.')
+    reply_to_target = fields.Selection([
+        ('original_discussion', 'Original Discussion'),
+        ('another_email_address', 'Another Email Address')],
+        string='Reply-To', default='original_discussion',
         help='Answers do not go in the original document discussion thread. This has an impact on the generated message-id.')
```
Comment (dbeguin) [inline addons/mail/wizard/mail_compose_message.py:134]: Update to help to match the selections : 
original_discussion = > Explain what it does exactly and what is the impact
another_email_address => Explain what it does exactly and what is the impact

File: addons/mail/models/mail_message.py [hunk 2]
```diff
@@ -114,7 +114,7 @@ def default_get(self, fields):
         help='Tracked values are stored in a separate model. This field allow to reconstruct '
              'the tracking and to generate statistics on the model.')
     # mail gateway
-    no_auto_thread = fields.Boolean(
+    check_reply_to = fields.Boolean(
         'No threading for answers',
         help='Answers do not go in the original document discussion thread. This has an impact on the generated message-id.')
```
Comment (dbeguin) [inline addons/mail/models/mail_message.py:119]: update name and help accordingly

File: addons/mail/wizard/mail_compose_message.py (hunk 1 above)
Comment (sht-odoo) [reply addons/mail/wizard/mail_compose_message.py]: > Update the help to match the selections :
> original_discussion = > Explain what it does exactly and what is the impact
> another_email_address => Explain what it does exactly and what is the impact

Done !!

File: addons/mail/models/mail_message.py (hunk 2 above)
Comment (sht-odoo) [reply addons/mail/models/mail_message.py]: Done !!

**Comment by seb-odoo** [comment]
I don't really know this part of mail/out of scope of what we do, so I'll let you guys fully handle it in @odoo/rd-sm

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-merge

**Pull request closed without merging**